pkg/gadgets/trace/capabilities/tracer/tracer_test.go

```diff
@@ -0,0 +1,421 @@
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil,
+		func(types.Event) {})
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+	t.Cleanup(tracer.Stop)
+}
+
+func TestTraceCapabilitiesTracerStopIdempotent(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil,
+		func(types.Event) {})
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+
+	// Check that a double stop doesn't cause issues
+	tracer.Stop()
+	tracer.Stop()
+}
+
+func TestCapabilitiesTracer(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+	// Needs kernel >= 5.1.0 because it introduced the InsetID field.
+	utilstest.RequireKernelVersion(t, &kernel.VersionInfo{Kernel: 5, Major: 1, Minor: 0})
+
+	false_ := false
+
+	type test struct {
+		getTracerConfig func(info *utilstest.RunnerInfo) *tracer.Config
+		runnerConfig    *utilstest.RunnerConfig
+		generateEvent   func() error
+		validateEvent   func(t *testing.T, info *utilstest.RunnerInfo, _ interface{}, events []types.Event)
+	}
+
+	for name, test := range map[string]test{
+		"captures_all_events_with_no_filters_configured": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{}
+			},
+			generateEvent: chown,
+			validateEvent: utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, _ interface{}) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					MountNsID: info.MountNsID,
+					Pid:       uint32(info.Pid),
+					UID:       uint32(info.UID),
+					Comm:      info.Comm,
+					CapName:   "CAP_CHOWN",
+					Cap:       0,
+					Audit:     1,
+					InsetID:   &false_,
+					Verdict:   "Allow",
+				}
+			}),
+		},
+		"captures_no_events_with_no_matching_filter": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, 0),
+				}
+			},
+			generateEvent: chown,
+			validateEvent: utilstest.ExpectNoEvent[types.Event, interface{}],
+		},
+		"captures_events_with_matching_filter": {
+			getTracerConfig: func(info *utilstest.RunnerInfo) *tracer.Config {
+				return &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+			},
+			generateEvent: chown,
+			validateEvent: utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, _ interface{}) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					MountNsID: info.MountNsID,
+					Pid:       uint32(info.Pid),
+					UID:       uint32(info.UID),
+					Comm:      info.Comm,
+					CapName:   "CAP_CHOWN",
+					Cap:       0,
+					Audit:     1,
+					InsetID:   &false_,
+					Verdict:   "Allow",
+				}
+			}),
+		},
+		"CAP_CHOWN": {
```

Comment:
Is it a duplicate from the `captures_events_with_matching_filter` test above?

Reply:
Yes, you're right. I removed it and rename other cases.